src/zarr/storage/_memory.py

```diff
@@ -38,7 +38,7 @@ class MemoryStore(Store):
 
     supports_writes: bool = True
     supports_deletes: bool = True
-    supports_partial_writes: bool = True
+    supports_partial_writes: bool = False
```

Comment:
I'm realizing this may be wrong. I interpreted `supports_partial_writes` to refer to whether `set_partial_values` is implemented whereas it may actually refer to whether `set` includes a `byte_range` parameter. I am not sure which, if either, is correct because `MemoryStore` had `supports_partial_writes=True` with `byte_range` implemented for `set` but no implemented `supports_partial_writes` whereas LocalStore had `supports_partial_writes=True` with no `byte_range` implemented for `set` but included a `supports_partial_writes` implementation.